server/pulp/server/db/migrations/lib/utils.py

```diff
@@ -0,0 +1,71 @@
+"""
+Utils for migrations.
+"""
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+MIGRATION_INIT_MSG = \
+    '* NOTE: This migration may take some time depending on the size of your Pulp content. *'
+MIGRATION_START_MSG = '* Migrating %s content...'
+MIGRATION_PROGRESS_MSG = '* Migrated units: %s of %s'
+
+
+class MigrationProgressLog(object):
+    """
+    Logs every 10% of migration completion.
+
+    :ivar migrated_units: number of already migrated units
+    :type migrated_units: int
+    :ivar content_type: name of the content to be migrated
+    :type content_type: str
+    :ivar total_units: total number of units to be migrated
+    :type total_units: int
+    """
+    migrated_units = 0
+    content_type = ''
+    total_units = 0
+
+    def __init__(self):
+        """
+        Logs header indicating start of migration.
+        Expected to be called before migration process begins.
+        """
+        stars = '*' * len(MIGRATION_INIT_MSG)
+        _logger.info(stars)
+        _logger.info(MIGRATION_INIT_MSG)
+        _logger.info(stars)
+
+    def start(self, content_type, total_units, msg=MIGRATION_START_MSG):
```

Comment:
This is begging to be a context manager! ;)

I think if you just create an ``__enter__`` and ``__exit__`` method that call ``start`` and ``stop`` respectively, that'll do it.